cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/service/ConnectionHandler.java

```diff
@@ -135,4 +164,153 @@ public void deleteConnection(HttpServiceRequest request, HttpServiceResponder re
       responder.sendStatus(HttpURLConnection.HTTP_OK);
     });
   }
+
+  @POST
+  @Path(API_VERSION + "/contexts/{context}/connections/test")
+  public void testConnection(HttpServiceRequest request, HttpServiceResponder responder,
+                             @PathParam("context") String namespace) {
+    respond(namespace, responder, namespaceSummary -> {
+      if (namespaceSummary.getName().equalsIgnoreCase(NamespaceId.SYSTEM.getNamespace())) {
+        responder.sendError(HttpURLConnection.HTTP_BAD_REQUEST,
+                            "Creating connection in system namespace is currently not supported");
+        return;
+      }
+
+      ConnectionCreationRequest creationRequest =
+        GSON.fromJson(StandardCharsets.UTF_8.decode(request.getContent()).toString(), ConnectionCreationRequest.class);
+
+      PluginConfigurer pluginConfigurer = getContext().createPluginConfigurer(namespaceSummary.getName());
+      try (Connector connector = getConnector(pluginConfigurer, creationRequest.getPlugin())) {
+        connector.configure(pluginConfigurer);
+        SimpleFailureCollector failureCollector = new SimpleFailureCollector();
+        try {
+          connector.test(failureCollector);
+          failureCollector.getOrThrowException();
+        } catch (ValidationException e) {
+          responder.sendJson(e.getFailures());
+          return;
+        }
+      }
+
+      responder.sendStatus(HttpURLConnection.HTTP_OK);
+    });
+  }
+
+  /**
+   * Browse the connection on a given path.
+   */
+  @GET
```

Comment:
For consistency, I think it would be better if this is a post (like the sample endpoint), with everything that is currently query params being in the request body.

Then 'test', 'browse', and 'sample' are all operations you perform on a connection, with the request body containing all the request properties.